diagnostic_feedback/tests/integration/base_test.py

```diff
@@ -0,0 +1,166 @@
+from xblock.fields import String
+from xblockutils.base_test import SeleniumXBlockTest
+from xblockutils.resources import ResourceLoader
+# Studio adds a url_name property to each XBlock but Workbench doesn't.
+# Since we rely on it, we need to mock url_name support so it can be set via XML and
+# accessed like a normal field.
+from diagnostic_feedback.quiz import QuizBlock
+
+QuizBlock.url_name = String()
+
+loader = ResourceLoader(__name__)
+
+
+class DiagnosticFeedbackBaseTest(SeleniumXBlockTest):
+    """
+    Base class for integration tests.
+    Scenarios can be loaded and edited on the fly.
+    """
+
+    def _get_previous_button(self):
+        return self.browser.find_elements_by_xpath("//ul[@role='menu']/li")[0]
+
+    def _get_previous_button_link(self):
+        return self.browser.find_element_by_xpath("//ul[@role='menu']/li/a[@href='#previous']")
+
+    def _get_next_button(self):
+        return self.browser.find_elements_by_xpath("//ul[@role='menu']/li")[1]
+
+    def _get_next_button_link(self):
+        return self.browser.find_element_by_xpath("//ul[@role='menu']/li/a[@href='#next']")
+
+    def wait_until_has_class_disabled(self, elem):
+        """ Need to wait until the given element has disabled class """
+
+        wait = WebDriverWait(elem, self.timeout)
+        wait.until(lambda e: 'disabled' in elem.get_attribute('class'),
+                   u"{} should be disabled".format(elem.text))
+
+    def _get_final_result_text(self):
+        return self.browser.find_element_by_css_selector('div.response_body').text
+
+    def _get_final_results(self):
+        return self.browser.find_elements_by_css_selector('div.response_body div.result')
+
+    def load_scenario(self, xml_file, params=None, load_immediately=True, view='student_view'):
+        """
+        Given the name of an XML file in the xml_templates folder, load it into the workbench.
+        """
+        params = params or {}
+        scenario = loader.render_template("xml_templates/{}".format(xml_file), params)
+        self.set_scenario_xml(scenario)
+        if load_immediately:
+            return self.go_to_view(view)
+
+    def load_student_view(self, xml_file, params=None):
+        """
+        Load student view in workbench
+        """
+        self.load_scenario("student_{}.xml".format(xml_file), params=params)
+
+    def load_studio_view(self, xml_file, params=None):
+        """
+        Load studio view in workbench
+        """
+        self.load_scenario("studio_{}.xml".format(xml_file), params=params, view='studio_view')
+
+
+class StudentBaseTest(DiagnosticFeedbackBaseTest):
+
+    def _get_startover_button(self):
+        return self.browser.find_elements_by_xpath("//ul[@role='menu']/li")[3]
+
+    def _get_startover_button_link(self):
+        return self.browser.find_element_by_xpath("//ul[@role='menu']/li/a[@href='#cancel']")
+
+    def _get_visible_choices(self):
+        """
+        Return one OR all visible choices
```

Comment:
@jareerahsan I think this docstring is a bit misleading. Why the emphasis on "or"? This method always returns a list of visible choices; callers can handle the return value the same way irrespective of the number of elements in the list. You could just go with "Return visible choices" here.